### DISCUSSION TITLE
Is there a way to "actually" modify the noise statistics of imu?

### DISCUSSION THREAD

**Discussion by skykim0609** [microsoft/AirSim#4057] [Support Q&A]
I am currently using AirSim with c++ api  to simulate visual inertial odometry algorithms.
I changed the noise parameters of IMU in settings.json, but turned out that the actual noise characterstics haven't changed.
I don't know the internal structure of how the IMU noise parameters are incorporated when simulating IMU in Airsim, but changing settings.json doesn't work(Even with 0.00 gyro & acceleration noise, the output is very noisy)

Is there any realistic way to solve this problem?

I am using AirSim 1.5.0 with Unreal 4.25.4

**Comment by jonyMarino** [comment]
Hi @skykim0609! There was an IMU fix we needed to revert due to another issue with PX4:
https://github.com/microsoft/AirSim/pull/3801
Maybe that is the issue you are facing?

**Reply by skykim0609** [reply]
The coordinate frame inversion is not the problem currently I am facing. The problem I am facing is that it seems like the noise characteristics seems like the simulator doesn't take account for the noise settings I wrote on settings.json file. (Or is the "ground truth" IMU data already contaminated from the beginning?)

**Reply by jonyMarino** [reply]
No. Ground truth data doesn't have noise.
